smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/waiters/JMESPathVisitor.kt

```diff
@@ -0,0 +1,232 @@
+    override fun visitComparator(expression: ComparatorExpression): String {
+        val left = expression.left!!.accept(this)
+        val right = expression.right!!.accept(this)
+        return addTempVar("comparison", "JMESValue(\$L) \$L JMESValue(\$L)", left, expression.comparator, right)
+    }
+
+    override fun visitCurrentNode(expression: CurrentExpression): String {
+        throw Exception("Unexpected current expression outside of flatten expression: $expression")
+    }
+
+    override fun visitExpressionType(expression: ExpressionTypeExpression): String {
+        throw Exception("ExpressionTypeExpression is unsupported")
+    }
+
+    override fun visitField(expression: FieldExpression): String {
+        return subfield(expression, "current")
+    }
+
+    // Filters elements from a collection which don't passing a test provided in a JMESPath child expression.
+    override fun visitFilterProjection(expression: FilterProjectionExpression): String {
+        val leftName = expression.left!!.accept(this)
+        val filteredName = bestTempVarName("${leftName}Filtered")
+        writer.openBlock("let \$L = Optional.some((\$L ?? []).filter { current in", "})", filteredName, leftName) {
+            writer.write("let current = Optional.some(current)")
+            val comparisonName = childBlock(expression.comparison!!)
+            writer.write("return \$L", comparisonName)
+        }
+        val right = expression.right!!
+        return mappingBlock(right, filteredName)
+    }
+
+    // Returns the inner expression unchanged when the inner expression is an array of non-array elements.
+    // Returns the inner expression flattened when the inner expression is an array of arrays.
+    override fun visitFlatten(expression: FlattenExpression): String {
+        val innerName = expression.expression!!.accept(this)
+        return addTempVar("${innerName}OrEmpty", "Optional.some((\$L ?? []).flattenIfPossible { $$0 })", innerName)
+    }
+
+    // Implement contains() and length() free functions which are the only 2 JMESPath methods we support.
+    // contains() returns true if its 1st param is a collection that contains an element equal
+    // to the 2nd param, false otherwise.
+    // length() returns the number of elements of an array, the number of key/value pairs for a map,
+    // or the number of characters for a string.
+    override fun visitFunction(expression: FunctionExpression): String {
+        when (expression.name) {
+            "contains" -> {
+                if (expression.arguments.size != 2) { throw Exception("Unexpected number of arguments to $expression") }
+
+                val subject = expression.arguments[0]
+                val subjectName = subject.accept(this)
+                val search = expression.arguments[1]
+                val searchName = search.accept(this)
+                return addTempVar("contains", "\$L.flatMap { \$L?.contains($$0) } ?? false", searchName, subjectName)
```

Comment:
I didn't see an example of this case in the `acceptors with generated code` branch. Can we make sure we have a test that hits this code path